### ISSUE TITLE
Return 404 instead of 405 in VC HTTP API

### ISSUE THREAD

**Issue by michaelsproul** [sigp/lighthouse#4596]
## Description

We fixed https://github.com/sigp/lighthouse/issues/3112 in the beacon node API but the same issue affects the validator client API. See that issue and the BN fix for details: https://github.com/sigp/lighthouse/pull/3836

An example of a request that triggers the error:

```fish
curl -s -H "Authorization: Bearer "(cat token.txt) "http://localhost:5062/not_found" | jq
```

```json
{
  "code": 405,
  "message": "METHOD_NOT_ALLOWED",
  "stacktraces": []
}
```

**Comment by ameya-deshmukh** [comment]
Taking this up!

**Comment by Gua00va** [comment]
@michaelsproul , I would like to take up this one!

**Comment by michaelsproul** [comment]
@ameya-deshmukh Did you end up implementing this already?

**Comment by ameya-deshmukh** [comment]
Yes! Opening a PR today/tomorrow.

**Comment by aoikurokawa** [comment]
@ameya-deshmukh  are you still working on this ? If not i would love to take it on

**Comment by ameya-deshmukh** [comment]
Hey @aoikurokawa I am! Will let you know if I don't find the time, but I am expecting it to be done soon. Sorry for the delay @michaelsproul, hope it's fine.

**Comment by duguorong009** [comment]
Hey, @ameya-deshmukh 
I am really sorry to open PR without any notice.
But, you haven't replied to my Discord friend request. 
And I found this issue is not that hard, and have good starting points.
So, I ended up creating PR in some hours.

@michaelsproul 
Can you review the PR #4758 & give the comments?
Thx in advance!

**Comment by ameya-deshmukh** [comment]
Hey no worries! Didn't get the time to work on this unfortunately :(
Thanks for taking it up :)

**Comment by chong-he** [comment]
Closing this as this is solved in #4758
